File: old/web/web-frontend/src/app/model/step/operation/CheckComposedStepDefUsageResponse.ts
import {Path} from "../../infrastructure/path/path.model";
import {Serializable} from "../../infrastructure/serializable.model";

export class CheckComposedStepDefUsageResponse implements Serializable<CheckComposedStepDefUsageResponse>{

    isUsed: boolean;
    isUniqueStepPattern: boolean;
    pathsForAffectedTests: Array<Path> = [];
    pathsForDirectParentSteps: Array<Path> = [];
    pathsForTransitiveParentSteps: Array<Path> = [];

    deserialize(input: Object): CheckComposedStepDefUsageResponse {
        this.isUsed = input["isUsed"];

        for (let pathsForParentTest of (input['pathsForParentTests']) || []) {
            this.pathsForAffectedTests.push(Path.deserialize(pathsForParentTest));
        }

        for (let pathsForDirectParentStep of (input['pathsForDirectParentSteps']) || []) {
            this.pathsForDirectParentSteps.push(Path.deserialize(pathsForDirectParentStep));
        }

        for (let pathsForTransitiveParentStep of (input['pathsForTransitiveParentSteps']) || []) {
            this.pathsForTransitiveParentSteps.push(Path.deserialize(pathsForTransitiveParentStep));
        }

        return this;
    }

    serialize(): string {
        throw new Error("Not implemented method");
    }
}

File: old/model/src/main/kotlin/com/testerum/model/step/operation/response/CheckComposedStepDefUsageResponse.kt
package com.testerum.model.step.operation.response

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path

data class CheckComposedStepDefUsageResponse @JsonCreator constructor(
    @JsonProperty("isUsed") val isUsed: Boolean = true,
    @JsonProperty("pathsForParentTests") val pathsForParentTests: List<Path> = emptyList(),
    @JsonProperty("pathsForDirectParentSteps") val pathsForDirectParentSteps: List<Path> = emptyList(),
    @JsonProperty("pathsForTransitiveParentSteps") val pathsForTransitiveParentSteps: List<Path> = emptyList()
)
